qemu/domain.go

```diff
@@ -372,6 +372,14 @@ func (d *Domain) Events() (chan qmp.Event, chan struct{}, error) {
 	// handle disconnection
 	go func() {
 		<-done
+		// drain anything that gets sent on the channel
+		// because the disconnect won't be processed if the
+		// listenAndServe loop is waiting for the listener
+		// to read from the unbuffered channel.
+		go func() {
+			for range stream {
```

Comment:
this can be simplified to `<-stream`, as a closed channel will always return

Reply:
I'm trying to decide whether there's any way for the loop to decide to process something else on that channel that's been waiting to go in before the disconnect... because that will only allow for one thing to be pulled off before needing the disconnect to happen.

Reply:
er, sorry. I missed some context, ignore that ^

However, why do this in a goroutine instead of draining inline prior to closing?

Reply:
if there are no messages waiting, neither <-stream nor range stream will return

Reply:
aye, lgtm. slow today.